[BUG] Unable to build rapids-4-spark jar from source due to missing 3.0.3-SNAPSHOT for spark-sql

**Describe the bug**
I'm trying to build the rapids-4-spark jar from v21.06.1 tag and am getting the following error:

```
[ERROR] Failed to execute goal on project rapids-4-spark-shims-spark303_2.12: Could not resolve dependencies for project com.nvidia:rapids-4-spark-shims-spark303_2.12:jar:21.06.1: Could not find artifact org.apache.spark:spark-sql_2.12:jar:3.0.3-SNAPSHOT in snapshots-repo (https://oss.sonatype.org/content/repositories/snapshots)
```

Not sure if this why, but looks like there isn't a snapshot for 3.0.3 in the apache repo (modified yesterday) - [https://repository.apache.org/content/repositories/snapshots/org/apache/spark/spark-sql_2.12/](https://repository.apache.org/content/repositories/snapshots/org/apache/spark/spark-sql_2.12/)

I got it to build by removing spark303 module from the pom, but just wanted to check with the devs here if this is expected.


**Steps/Code to reproduce bug**
Ran the following command from root directory for tag v21.06.1:
`mvn -DskipTests package`

**Expected behavior**
Jar should be built without errors.

**Environment details**
 - OS: Linux
 - Java version: 1.8.0_292
 - Maven version: 3.8.1


**Additional context**
I am need to build a local jar to remove the SLF4j dependency that rapids jar brings with it (v1.7.30), I'm running into issues with LoggerFactory instantiation in my environment, possibly due to a different library bringing its own slf4j (v1.7.16).

```
Failed to instantiate SLF4J LoggerFactory
Reported exception:
java.lang.NullPointerException
at org.slf4j.LoggerFactory.reportActualBinding(LoggerFactory.java:349)
```
Any guidance here that could help unblock me (that wouldn't require me to remove these dependencies from source and re-build) would also be really appreciated. Thank you in advance!

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -40,6 +40,14 @@ mvn verify
 
 After a successful build the RAPIDS Accelerator jar will be in the `dist/target/` directory.
 
+### Building Shims for Spark Snapshot Versions
+
+By default the build will only include shims for released versions of Spark. To include shims
+for snapshot versions of Spark still under development, use the `snapshot-shims` Maven profile
+(e.g.: add `-Psnapshot-shims` to the Maven command-line). Note that when a snapshot Spark version
+later becomes an official release, the snapshot shim for that version may no longer build due to
+missing snapshot artifacts for that Spark version.
+
 ### Building against different CUDA Toolkit versions
 
 You can build against different versions of the CUDA Toolkit by using one of the following profiles:

--- a/shims/aggregator/pom.xml
+++ b/shims/aggregator/pom.xml
@@ -98,7 +98,7 @@
         <profile>
             <id>snapshot-shims</id>
             <activation>
-                <activeByDefault>true</activeByDefault>
+                <activeByDefault>false</activeByDefault>
             </activation>
             <dependencies>
                 <dependency>

--- a/shims/pom.xml
+++ b/shims/pom.xml
@@ -53,7 +53,7 @@
         <profile>
             <id>snapshot-shims</id>
             <activation>
-                <activeByDefault>true</activeByDefault>
+                <activeByDefault>false</activeByDefault>
             </activation>
             <modules>
                 <module>spark304</module>

